channels/rdpgfx/client/rdpgfx_main.c

```diff
@@ -159,7 +164,9 @@ static UINT rdpgfx_send_caps_advertise_pdu(RdpgfxClientContext* context,
 
 static BOOL rdpgfx_is_capability_filtered(RDPGFX_PLUGIN* gfx, UINT32 caps)
 {
-	const UINT32 filter = gfx->capsFilter;
+	WINPR_ASSERT(gfx);
+	const UINT32 filter =
+	    freerdp_settings_get_uint32(gfx->rdpcontext->settings, FreeRDP_GfxCapsFilter);
```

Comment:
Don't we have a `gfx->settings` field shortcut that we could use here and below ?

Reply:
@hardening removed that here, as I try to unify (and eliminate) the duplicated fields (makes changes to `rdpContext` very error prone if we store the pointers everywhere but never update them)